src/DetailView/DetailView.js

```diff
@@ -38,25 +41,38 @@ class DetailView extends ConnectionComponent {
 	componentWillUnmount() {
 		this.unregisterAllNotifications();
 	}
+
+	handleEventChildrenSwitch(event, isInputChecked) {
+		this.includeEventChildren = isInputChecked;
+		this.props.refreshEventTypes(this.handleEventTypeChange, isInputChecked);
+	}
 	
 	handleServerSideEventsChanged(eventTypeId) {
 		if (this.props.eventTypes.value.length) {
-			this.props.lazyEventLoading(this.state.currentEventType.Id, this.handleEventChange);
+			this.props.lazyEventLoading(
+				this.state.currentEventType.Id,
+				this.handleEventChange,
+				this.includeEventChildren
+			);
 		}
-		if(eventTypeId && this.props.eventTypes.value.find((eventType) => {
+		if (eventTypeId && this.props.eventTypes.value.find((eventType) => {
 				return eventType.Id === eventTypeId;
 			}) === undefined) {
-			this.props.refreshEventTypes(this.handleEventTypeChange);
+			this.props.refreshEventTypes(this.handleEventTypeChange, this.includeEventChildren);
 		}
 	}
 
-	handleEventTypeChange(eventType) {
-		this.setState({
-			currentEventType: eventType,
-			filter: [],
-		});
-		this.props.lazyAttributeLoading(eventType.Id);
-		this.props.lazyEventLoading(eventType.Id, this.handleEventChange);
+	handleEventTypeChange(eventTypes) {
+		if (eventTypes && eventTypes.length > 0) {
+			this.setState({
+				currentEventType: eventTypes[0],
```

Comment:
You _could_ also save `eventTypes[0]` in a `const` variable as you are accessing it thrice.